www/nginx/src/opnsense/service/templates/OPNsense/Nginx/location.conf

```diff
@@ -119,13 +119,17 @@ location {{ location.matchtype }} {{ location.urlpattern }} {
 {%   endif %}
 {% endif%}
 {% if location.upstream is defined and (location.php_enable is not defined or location.php_enable != '1') %}
-{%   set upstream = helpers.getUUID(location.upstream) %}
-    proxy_set_header Host $host;
+{%   set upstream = helpers.getUUID(location.upstream) %} 
+{% if location.proxy_add_port_to_header is defined and  location.proxy_add_port_to_header = '1' %}
+   proxy_set_header Host $host:$server_port;
```

Comment:
this is incorrect. The host header is meant to contain the header which is used by the next server to determine which virtual host it should use. so I don't think this belongs here. as it should contain the port of the backend server. Not the client port:
https://www.rfc-editor.org/rfc/rfc2616#section-14.23

Reply:
Isn't server_port the backend port?
If so, we can use the correct var in here.

Reply:
no, see http://nginx.org/en/docs/http/ngx_http_core_module.html